crates/kornia-tensor/src/storage.rs

```diff
@@ -340,12 +340,63 @@ mod tests {
     #[test]
     fn test_tensor_mutability() -> Result<(), TensorAllocatorError> {
         let vec: Vec<i32> = vec![1, 2, 3, 4, 5];
-        let mut buffer = TensorStorage::<_, CpuAllocator>::from_vec(vec, CpuAllocator);
+        let mut buffer = TensorStorage::<_, CpuAllocator>::from_vec(vec, CpuAllocator::default());
         let ptr_mut = buffer.as_mut_ptr();
         unsafe {
             *ptr_mut.add(0) = 10;
         }
         assert_eq!(buffer.into_vec(), vec![10, 2, 3, 4, 5]);
         Ok(())
     }
+
+    #[test]
+    fn test_parent_deallocator() -> Result<(), TensorAllocatorError> {
+        use std::sync::atomic::{AtomicBool, Ordering};
+        use std::sync::Arc;
+
+        #[derive(Clone)]
+        struct TestParentDeallocator {
```

Comment:
maybe use mockall or something similar, there are mock test frameworks that can trace calls like this